main.go

```diff
@@ -0,0 +1,139 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+)
+
+type Record struct {
+	ID        uint
+	FirstName string
+	LastName  string
+	PhoneN    string
+}
+
+func removeEntry() (errr string) {
+	var Records []Record
+	f, err := os.Open("Data.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
```

Comment:
nit: A funny part about this particular `defer` is that you gonna close the file too late. It is better to move the logic which reads the file (and immediately closes it) to a dedicated function e.g.

```go
func loadPhonebook() (r []Record, err error) {
  f, err := os.Open(filename)
  if err != nil {
    return nil, err
  }
  defer f.Close()
  if err := json.NewDecoder(f).Decode(&r); err != nil {
    return nil, err
  }
  return r, nil
}

func removeEntry() (err error) {
  entries, err := loadPhonebook()
  if err != nil {
    return err
  }
  ...
}
```
  
`loadPhonebook` guarantees the file is closed as soon as it isn't required anymore